core/Tracker/RequestSet.php

```diff
@@ -244,12 +244,14 @@ public function restoreEnvironment()
     private function resetEnvironment($env)
     {
         $_SERVER = $env['server'];
+        $_COOKIE = $env['cookie'];
```

Comment:
I would probably add a check `if (isset($env['cookie'])) {}` just in case it is not set for possibly still queued entries that do not have the cookie set ye